cpp/src/phonenumbers/phonenumbermatcher.cc

```diff
@@ -421,12 +422,26 @@ PhoneNumberMatcher::PhoneNumberMatcher(const string& text,
       max_tries_(numeric_limits<int>::max()),
       state_(NOT_READY),
       last_match_(NULL),
-      search_index_(0) {
+      search_index_(0),
+      isInputInCleanUTF8_(true) {
+  isInputInCleanUTF8_ =  CheckInputUTF8OrNot();
 }
 
+
 PhoneNumberMatcher::~PhoneNumberMatcher() {
 }
 
+// static
+bool PhoneNumberMatcher::CheckInputUTF8OrNot() {
+  // Input should contain only UTF-8 characters.
```

Comment:
Comment would be better if it explained *why* -> but then again, I wouldn't put it here, I would put it where you use the variable. Here the code doesn't need a comment IMO.

Reply:
Done. Thanks Lara :)